packages/client/src/Stream.ts

```diff
@@ -108,15 +108,19 @@ class StreamrStream implements StreamMetadata {
     /**
      * Persist stream metadata updates.
      */
-    async update() {
+    async update(props: Omit<StreamProperties, 'id'>) {
+        for (const key of Object.keys(props)) {
+            // @ts-expect-error
+            this[key] = props[key]
+        }
```

Comment:
Since updating the stream can fail due to smart contract issues, I think we should update the object itself only after we know the the transaction went thru successfully?

There should also be a test case for this.

Reply:
Fixed and added a test